dotCMS/src/main/java/com/dotmarketing/factories/PublishFactory.java

```diff
@@ -165,6 +188,74 @@ public static boolean publishAsset(Inode webAsset, User user, boolean respectFro
 		return publishAsset(webAsset,user,respectFrontendRoles, true);
 
 	}
+
+	/**
+	 * This method publish a given and asset and its related assets
+	 * if a user is passed the method will check permissions to publish
+	 * if the user doesn't have permission to publish one of the related assets
+	 * then that one will be skipped
+	 * @param folder
+	 * @param user
+	 * @param respectFrontendRoles
+	 * @param isNewVersion  - if passed false then the webasset's mod user and mod date will NOT be altered. @see {@link ContentletAPI#checkinWithoutVersioning(Contentlet, java.util.Map, List, List, User, boolean)}checkinWithoutVersioning.
+	 * @return
+	 * @throws WebAssetException
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+	@SuppressWarnings("unchecked")
+	public static boolean publishAsset(Folder folder, User user, boolean respectFrontendRoles,
+			boolean isNewVersion) throws WebAssetException, DotSecurityException, DotDataException {
+		ContentletAPI conAPI = APILocator.getContentletAPI();
+
+		//http://jira.dotmarketing.net/browse/DOTCMS-6325
+		if (user != null &&
+				!permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_EDIT,
+						user)) {
+			Logger.debug(PublishFactory.class, "publishAsset: user = " + user.getEmailAddress()
+					+ ", don't have permissions to publish: " + folder);
+			return false;
+		}
+
+		Logger.debug(PublishFactory.class, "*****I'm a Folder -- Publishing" + folder.getName());
```

Comment:
use supplier version to avoid the memory creation when logger debug is off

Reply:
Done